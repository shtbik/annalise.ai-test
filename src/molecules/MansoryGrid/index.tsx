import { FC } from 'react'

import { StyledComponent } from '../../types/component'

import { StyleGrid } from './styled.index'

export interface IMansoryGrid extends StyledComponent {
  columnCount: number
  columnGap?: number
}

export const defaultColumnGap = 10

const MansoryGrid: FC<IMansoryGrid> = ({ children, columnCount, columnGap = defaultColumnGap, className }) => {
  return (
    <StyleGrid $columnCount={columnCount} $columnGap={columnGap} className={className}>
      {children}
    </StyleGrid>
  )
}

export default MansoryGrid
