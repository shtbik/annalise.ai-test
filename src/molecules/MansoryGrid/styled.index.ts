import styled from 'styled-components'

interface IStyleGrid {
  $columnCount: number
  $columnGap: number
}

export const StyleGrid = styled.div<IStyleGrid>`
  column-count: ${({ $columnCount }) => $columnCount};
  column-gap: ${({ $columnGap }) => `${$columnGap}px`};

  > * {
    display: grid;
    grid-template-rows: 1fr auto;
    overflow: hidden;
    break-inside: avoid;
  }
`
