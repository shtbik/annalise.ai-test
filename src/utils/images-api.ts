import { createApi } from 'unsplash-js'

const { REACT_APP_UNSPLASH_API_URL } = process.env

const origin = REACT_APP_UNSPLASH_API_URL || `http://${window.location.hostname}:8080`

const unsplashApi = createApi({
  apiUrl: `${origin}/unsplash-proxy`,
})

export default unsplashApi
