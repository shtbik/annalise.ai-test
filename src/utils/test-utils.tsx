import { FC, ReactElement } from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { ThemeProvider } from 'styled-components'

import { render, RenderOptions } from '@testing-library/react'

import theme from '../styles/theme'

const AllTheProviders: FC = ({ children }) => {
  return (
    <Router>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </Router>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
