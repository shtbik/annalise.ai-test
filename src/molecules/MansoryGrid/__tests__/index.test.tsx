import styled from 'styled-components'

import { render } from '../../../utils/test-utils'

import MansoryGrid, { defaultColumnGap } from '../'

const SMansoryGrid = styled(MansoryGrid)`
  margin: 10px;
`

const columnCount = 4
const childrens = (
  <>
    <div>1</div>
    <div>2</div>
    <div>3</div>
    <div>4</div>
  </>
)

describe('MansoryGrid component', () => {
  it('default props', () => {
    const { container } = render(<MansoryGrid columnCount={columnCount}>{childrens}</MansoryGrid>)

    expect(container.firstChild).toHaveStyleRule(`column-gap`, `${defaultColumnGap}px`)
    expect(container.firstChild).toHaveStyleRule(`column-count`, columnCount.toString())
  })

  it('snapshot', () => {
    const { container } = render(
      <SMansoryGrid columnCount={6} columnGap={15}>
        {childrens}
      </SMansoryGrid>
    )
    expect(container).toMatchSnapshot()
  })
})
