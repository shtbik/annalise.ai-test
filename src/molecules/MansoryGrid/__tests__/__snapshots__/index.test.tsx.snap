// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MansoryGrid component snapshot 1`] = `
.c0 {
  -webkit-column-count: 6;
  column-count: 6;
  -webkit-column-gap: 15px;
  column-gap: 15px;
}

.c0 > * {
  display: grid;
  grid-template-rows: 1fr auto;
  overflow: hidden;
  -webkit-break-inside: avoid;
  break-inside: avoid;
}

.c1 {
  margin: 10px;
}

<div>
  <div
    class="c0 c1"
  >
    <div>
      1
    </div>
    <div>
      2
    </div>
    <div>
      3
    </div>
    <div>
      4
    </div>
  </div>
</div>
`;
